src/cmd_parser.cc

```diff
@@ -95,9 +113,33 @@ CmdParser::CmdParser(int argc, const char* const argv[]) {
         getLastValueOrDefault(n_learning_epoch_, (num_epochs_t)0);
     n_inference_epoch =
         getLastValueOrDefault(n_inference_epoch_, (num_epochs_t)0);
+
     n_datacopy = getLastValueOrDefault(n_datacopy_, (size_t)0);
+    // ensure n_datacopy is a sane number
+    if (n_datacopy == 0) n_datacopy = NumaNodes::num_configured();
+    check(0 < n_datacopy && n_datacopy <= NumaNodes::num_configured())
+        << "n_datacopy (" << n_datacopy << ") must be in the range of [0, "
+        << NumaNodes::num_configured() << "]" << std::endl;
+    check(n_datacopy % NumaNodes::num_configured() == 0)
```

Comment:
Should this be the other way around? @netj 
i.e., `NumaNodes::num_configured() % n_datacopy == 0`